web/src/components/vis/PlsdaPage/PlsdaPage.vue

```diff
@@ -0,0 +1,227 @@
+<script>
+import VisTileLarge from '@/components/vis/VisTileLarge.vue';
+import LayoutGrid from '@/components/LayoutGrid.vue';
+import ScorePlot from './ScorePlot.vue';
+import LoadingsPlot from './LoadingsPlot.vue';
+import plotData from '../mixins/plotData';
+
+export default {
+  components: {
+    ScorePlot,
+    LoadingsPlot,
+    VisTileLarge,
+    LayoutGrid,
+  },
+
+  mixins: [plotData('plsda')],
+
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      pcXval: '1',
+      pcYval: '2',
+      numComponentsVal: '10',
+      pcX: 1,
+      pcY: 2,
+      numComponents: 10,
+      showEllipses: true,
+      showCrosshairs: true,
+      showCutoffs: true,
+      showScore: true,
+      showLoadings: true,
+    };
+  },
+
+  computed: {
+    ready() {
+      const pcaReady = this.$store.getters.ready(this.id, 'plsda_scores');
+      const loadingsReady = this.$store.getters.ready(this.id, 'plsda_loadings');
+      return pcaReady && loadingsReady;
+    },
+    loadings() {
+      if (this.plot?.data?.loadings) {
+        return this.plot.data.loadings;
+      }
+      return [];
+    },
+    pcCoords() {
+      if (this.plot?.data?.scores?.x) {
+        return this.plot.data.scores.x;
+      }
+      return [];
+    },
+    eigenvalues() {
+      if (this.plot?.data?.scores?.sdev) {
+        return this.plot.data.scores.sdev;
+      }
+      return [];
+    },
+    rowLabels() {
+      if (this.dataset?.row?.data) {
+        return this.dataset.row.data.filter((row) => row.row_type === 'sample').map((row) => row.row_name);
+      }
+      return [];
+    },
+    groupLabels() {
+      if (this.dataset?.validatedGroups?.data) {
+        const groupColumnName = this.dataset.validatedGroups.columnNames[0];
+        const groupLabels = {};
+        groupLabels[groupColumnName] = this.dataset.validatedGroups.data.map((group) => group);
+        return groupLabels;
+      }
+      return {};
+    },
+    columns() {
+      if (this.dataset?.column?.data) {
+        return this.dataset.column.data;
+      }
+      return [];
+    },
+
+    groupLevels() {
+      if (this.dataset?.groupLevels) {
+        return this.dataset.groupLevels;
+      }
+      return [];
+    },
+  },
+  watch: {
+    pcXval: {
+      handler(val) {
+        const pcX = Number.parseInt(val, 10);
+        if (!Number.isNaN(pcX)) {
+          this.pcX = pcX;
+        }
+      },
+      immediate: true,
+    },
+    pcYval: {
+      handler(val) {
+        const pcY = Number.parseInt(val, 10);
+        if (!Number.isNaN(pcY)) {
+          this.pcY = pcY;
+        }
+      },
+      immediate: true,
+    },
+    numComponentsVal: {
+      handler(val) {
+        const numComponents = Number.parseInt(val, 10);
+        if (!Number.isNaN(numComponents)) {
+          this.numComponents = numComponents;
+        }
+      },
+      immediate: true,
+    },
+  },
+};
```

Comment:
If you find yourself setting data inside a watcher, it's often the case you should use a computed prop instead.

Now I remember that, for some reason, @dchiquit wrote these this way because of https://github.com/girder/viime/pull/531

He said that if you backspace the input box, the vis will explode because you called `parseInt` on the empty string.  However, we now have 3 duplications of this code.  1 uses the watch-set style, the other `vis/scorePlotTile.vue` uses regular computed functions.  Both work.

Hoping @dchiquit can provide some insight about why his watch-update fix was necessary in `vis/PcaPage/PcaPage.vue` but not `vis/scorePlotTile.vue`.

Reply:
@dchiquit could you provide some input so we can get this closer to merging?